src/app/modules/form/public-form/public-form.middlewares.ts

```diff
@@ -0,0 +1,31 @@
+'use strict'
+
+import { RequestHandler } from 'express'
+import { StatusCodes } from 'http-status-codes'
+
+import { WithForm } from '../../../../types'
+import { FormDeletedError } from '../form.errors'
+import { isFormPublic } from '../form.service'
+
+/**
+ * Express middleware function that checks if a form attached to the Express request handler is public.
+ * before allowing downstream middleware to handle the request. Otherwise, it returns a HTTP 404 if the
+ * form is private, or HTTP 410 if the form has been archived.
+ * @param req - Express request object
+ * @param res - Express response object
+ * @param next - Express next middleware function
+ */
+export const isFormPublicCheck: RequestHandler = (req, res, next) => {
+  const { form } = req as WithForm<typeof req>
+  return isFormPublic(form)
+    .map(() => next())
+    .mapErr((error) => {
```

Comment:
question: do we want to log the errors here?

Reply:
I don't think so, as it can be a fairly common occurrence. It's more important to log exceptions rather than errors (although JavaScript doesn't distinguish between them explicitly unlike Java).

Reply:
oh ye, i see the difference